### PR TITLE
create a config loader similar to flask one

### PR DESCRIPTION
**Pull request by szymonlopaciuk**
Signed-off-by: Szymon Łopaciuk <szymon.lopaciuk@cern.ch>

### CODE REVIEW

**Comment by david-caro** [comment]
Sorry for the noise... I was creating the labels for waffle.

File: tests/test_config.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    config = Config(defaults={
+        'SERVER_NAME': '127.0.0.1',
+        'SOME_OTHER_DEFAULT': 1234,
+    })
+    config.load_pyfile(mock_config.strpath)
+
+    assert config['SERVER_NAME'] == '0.0.0.0'
+    assert config['OTHER_VARIABLE'] == 42
+    assert config.get('SOME_OTHER_DEFAULT') == 1234
+    assert config.get('NO_SUCH_VARIABLE', 'empty') == 'empty'
```
Comment (david-caro) [inline tests/test_config.py]: Probably better to test the non-existance of the key, rather than the value, I know that it's unlikely but this will pass the test:
```python
config = {
    'NO_SUCH_VARIABLE': 'empty',
}
```

File: tests/test_config.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""ORCID configuration loader."""
+
+from __future__ import absolute_import, division, print_function
+
```
Comment (david-caro) [inline tests/test_config.py]: Can you add a simple test with an empty config file, and another with a non-valid python file?

I'm curious what the errors are xd

File: inspire_utils/config.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+"""ORCID configuration loader."""
+
+from __future__ import absolute_import, division, print_function
+
+
+class Config(dict):
+    def __init__(self, defaults=None):
+        super(Config, self).__init__(defaults or {})
+
+    def load_pyfile(self, path):
```
Comment (david-caro) [inline inspire_utils/config.py]: I'd add a default path, so all the places this is used can "reuse" that default one. Probably as a global in this module for now.

Comment (david-caro) [reply inspire_utils/config.py]: Something like:
```
DEFAULT_CONF_PATH = os.path.join(os.curdir, 'inspirehep.cfg')

Comment (david-caro) [reply inspire_utils/config.py]: :/ though currently in prod/qa the path is `os.path.join(os.curdir, 'var/inspirehep-instance/inspirehep.cfg')`.

It might be better to add a function here, like `load_config` that does that for you, something like:
```python
DEFAULT_FILE_PATHS = (
    './var/inspirehep-instance/inspirehep.cfg',
    'inspirehep.cfg',
)


def load_config(paths=DEFAULT_FILE_PATHS):
    conf = Config()
    for path in paths:
        if os.path.isfile(path):
            conf.load_pyfile(path)

    return conf
```

Or similar (not 100% convinced by the above solution though).

File: inspire_utils/config.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""ORCID configuration loader."""
+
```
Comment (puntonim) [inline inspire_utils/config.py]: Pls add a comment like:
"Inspired by the Flask configuration loader: https://github.com/pallets/flask/blob/40745bb338c45498ca19010175f341332ab2eefb/flask/config.py"

**Comment by jacquerie** [comment]
Gah, semantic versioning broken! You should have added a `Sem-ver: adds new feature` trailer to the merge commit (like [here](https://github.com/inspirehep/inspire-utils/commit/b93aac5fe4ca69b86ff62ba4ee44336f6bc2439e)) to bump the version in the right way. No biggie!

**Comment by szymonlopaciuk** [comment]
Sorry :grimacing: Won't happen again
